### PR TITLE
Improve linter rules

### PR DESCRIPTION
**Pull request by FPierre**
* Add changelog file (explanations and examples when a rule changes)
* Improve README with config example
* Disable Cubyn rules for now (not ready yet)

Enjoy cleaner PR :rocket:

@getvega asked for multiple linters configurations (one for soft rules, one for harder rules) to manage legacy codes. Given the rules in this PR, I don't thing it is relevant anymore. What do you think about that?

After this PR merge, I will make you a PR with new rules to gives you an idea of the changes

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,66 +1,103 @@
...
+        // 'cubyn/meta-permissions': 'off',
+
+        //
+        // Airbnb rules
+        //
+
+        // Activations/updates
+
+        'func-style': ['error', 'declaration', { allowArrowFunctions: true }],
+        'id-length': ['error', { min: 3, properties: 'never' }],
```
Comment (getvega) [inline index.js]: setting a max would be nice as well

`expectedGridDurationMinValue` for instance is hard on my eyes. this usually denotes SRP violation. i.E:

```
function checkGridDuration(value) {
  const minValue = ...; // <-- the context enables shorter var names
}
```

File: index.js [hunk 2]
```diff
@@ -1,66 +1,103 @@
...
+        'line-comment-position': ['error', {
+          position: 'above',
+          ignorePattern: '',
+          applyDefaultPatterns: true,
+        }],
+        'max-depth': ['error', 4],
```
Comment (getvega) [inline index.js]: 👍

File: index.js [hunk 3]
```diff
@@ -1,66 +1,103 @@
...
+        'max-nested-callbacks': ['error', { max: 3 }],
+        'max-statements-per-line': ['error', { max: 1 }],
+        'newline-per-chained-call': ['error', { ignoreChainWithDepth: 2 }],
+        'no-eq-null': 'error',
+        'no-implicit-globals': 'error',
+        'no-magic-numbers': ['error', {
+          ignoreArrayIndexes: true,
+          ignore: [ -1, 0, 1, 2, 10],
+          enforceConst: true
```
Comment (getvega) [inline index.js]: omg!! 👍

File: index.js [hunk 4]
```diff
@@ -1,66 +1,103 @@
...
+            blankLine: 'always',
+            prev: ['const', 'let', 'var'],
+            next: '*',
+          },
+          {
+            blankLine: 'any',
+            prev: ['const', 'let', 'var'],
+            next: ['const', 'let', 'var', 'expression'],
+          },
+        ],
```
Comment (getvega) [inline index.js]: I feel this is something that will be v restrictive but I will have to try it.

File: index.js (hunk 1 above)
Comment (stombre) [reply index.js]: hard to define what is a "long" variable, sometime I prefer big variable name to explain an expression.
Exemple pourri ;
```
const isParcelUnvalidForAfgathisthan = parcel.countr === 'an' && parcel.itemsJson[0] === 'alcool';

if (isParcelUnvalidForAfgathisthan) {
```

**Pull request merged**